console/src/main.rs

```diff
@@ -15,6 +15,25 @@ mod tasks;
 mod term;
 mod view;
 
+enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+async fn connect(
```

Comment:
minor nit: i would generally put this below `main` --- i like to sort of topologically sort functions in a module so that major ones that call the others are at the top of the file, and the smaller helper functions that the main ones call are below them. IMO it makes more sense to read the file that way.

but this is not a big deal and i'd be happy to merge this as-is.